packages/scheduler/lib/scheduler.ts

```diff
@@ -104,52 +104,55 @@ export class Scheduler {
      * const scheduled = await scheduler.immediate(schedulingProps);
      */
     public async immediate(props: ImmediateProps | { scheduleName: string }): Promise<Result<Task>> {
-        let taskProps: tasks.TaskProps;
```

Comment:
diff is really confusing here. The change is just wrapping everything in a transaction and using `forUpdate` when calling `schedules.search`

Reply:
the trick is to hideWhitespaces changes, not sure why it's not the default
<img width="278" alt="Screenshot 2024-06-10 at 17 33 22" src="https://github.com/NangoHQ/nango/assets/1637651/3ba3af3e-55a2-4019-9c78-70f99782d40f">